nemoguardrails/library/autoguard/README.md

```diff
@@ -0,0 +1,165 @@
+
+```yaml
+rails:
+  config:
+    autoguard:
+      parameters:
+        endpoint: "http://35.225.99.81:8888/guardrail"
+      input:
+        guardrails:
+          - racial_bias_detection
+          - gender_bias_detection
+          - confidential_detection
+          - harm_detection
+          - text_toxicity_extraction
+          - jailbreak_detection
+      output:
+        guardrails:
+          - racial_bias_detection
+          - gender_bias_detection
+          - confidential_detection
+          - harm_detection
+          - text_toxicity_extraction
+          - jailbreak_detection
+```
+We also have to add the autoguard's endpoint in parameters.
+
+The colang file has to be in the following format:
+
+```colang
+define subflow output autoguard
+    $result = execute autoguard_api
+    if $result[0] == True
+        bot refuse to respond autoguard
+        stop
+
+define bot refuse to respond autoguard
+    "$result[1] has been detected by AutoGuard; Sorry, can't process."
+```
+
+### Confidential detection
+
+The goal of the confidential detection rail is to determine if the text has any kind of confidential information. This rail can be applied at both input and output. This guardrail can be added by adding `confidential_detection` in `autoguard` section in `config.yml`
+
+### Gender bias detection
+
+The goal of the gender bias detection rail is to determine if the text has any kind of gender biased content. This rail can be applied at both input and output. This guardrail can be added by adding `gender_bias_detection` in `autoguard` section in `config.yml`
+
+### Harm detection
+
+The goal of the harm detection rail is to determine if the text has any kind of harm to human content. This rail can be applied at both input and output. This guardrail can be added by adding `harm_detection` in `autoguard` section in `config.yml`
+
+### Toxicity detection
+
+The goal of the toxicity detection rail is to determine if the text has any kind of toxic content. This rail can be applied at both input and output.This guardrail can be added by adding `text_toxicity_extraction` in `autoguard` section in `config.yml`
+
+### Racial bias detection
+
+The goal of the racial bias detection rail is to determine if the text has any kind of racially biased content. This rail can be applied at both input and output.
+This guardrail can be added by adding `racial_bias_detection` in `autoguard` section in `config.yml`
+
+### Jailbreak detection
+
+The goal of the jailbreak detection rail is to determine if the text has any kind of jailbreak attempt.
+This rail can be applied at both input and output.This guardrail can be added by adding `jailbreak_detection` in `autoguard` section in `config.yml`
+
+## Usage (AutoGuard PII)
```

Comment:
@abhijitpal1247 how are the config and advanced config are set for PII and other guardrails? the full config can be found here:  https://github.com/decypher-ai/autoguard/wiki/Making-a-request-to-the-test-instance.

The PII guardrail has additional configs such as: enabled_types (list of PII types that should be redacted), "contextual_rules" (list of contextual rules, each consists of a list of pii types, and at least one rule bust e present), and  "matching_scores" (ranging from 0 to 1- for detected classes). Whereas other guardrails have additional "matching_scores" configurations. 
** Config is optional, if not provided, default deployment config will be used
** mode can be "OFF" "DETECT" "BLOCK"

Reply:
Hi Rahm, based on discussions with Kimi, currently the guardrails' config is like: https://jupyter.armilla.ai/notebooks/autoguard-internal.ipynb

Reply:
so we have not provided advanced config matching scores

Reply:
regarding "BLOCK", "DETECT" are same for "/guardrail" endpoint

Reply:
@abhijitpal1247 there are some additional optional params in that wiki, that we can include

Reply:
resolved in prev commits